src/components/editor/State.js

```diff
@@ -1176,10 +1177,52 @@ class AllergyOnset extends Component<Props> {
           <br />
           <Codes codes={state.codes} system={"SNOMED-CT"} onChange={this.props.onChange('codes')} />
         </div>
+        <div>
+          <label>Allergy Type:
+            <RIESelect className='editable-text'
+              propName='allergy_type'
+              value={{id: state.allergy_type, text: state.allergy_type}}
+              change={this.props.onChange('allergy_type')}
+              options={[{id:'allergy', text: 'Allergy'}, {id:'intolerance', text: 'Intolerance'}]}
+            />
+          </label>
+        </div>
+        <div>
+          <label>Category:
+            <RIESelect className='editable-text'
+              propName='category'
+              value={{id: state.category, text: state.category}}
+              change={this.props.onChange('category')}
+              options={[{id:'food', text: 'Food'},
+                        {id:'medication', text: 'Medication'},
+                        {id:'environment', text: 'Environment'}]}
+            />
+          </label>
+        </div>
+        <div className='section'>
+        Reactions:
+        {this.renderReactions(state)}
+        <a className='editable-text' onClick={() => this.props.onChange(`reactions[${this.props.state.reactions.length}]`)({val: {id: {reaction: {code: "1234", system: "SNOMED-CT", display: "SNOMED Code"}, possible_severities: [{level: "mild", value: 1}]}}})}>+</a>
+        </div>
       </div>
     );
   }
 
+  renderReactions(state) {
+    if(state.reactions == null) {
+      return null;
+    } else {
+      return state.reactions.map((r, i) => {
+          return (
+            <div className='section' key={i}>
+              <a className='editable-text delete-button' onClick={() => this.props.onChange(`reactions[${i}]`)({val: {id: null}})}>remove</a>
```

Comment:
When you hit remove it leaves a null in the JSON, which seems to cause issues.

This is also true for severities.

I know this problem exists for some other states as well, so this might be an endemic issue that we fix later.

E.g.

```json
{
  "name": "Untitled",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "direct_transition": "New_State"
    },
    "Terminal": {
      "type": "Terminal"
    },
    "New_State": {
      "allergy_type": "allergy",
      "category": "food",
      "type": "AllergyOnset",
      "target_encounter": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": "1234",
          "display": "SNOMED Code"
        }
      ],
      "reactions": [
        {
          "reaction": {
            "code": "1234",
            "system": "SNOMED-CT",
            "display": "SNOMED Code"
          },
          "possible_severities": [
            {
              "level": "mild",
              "value": 1
            },
            null
          ]
        },
        null
      ],
      "direct_transition": "Terminal"
    }
  },
  "gmf_version": 2
}
```